From: Dmitry Kasatkin <dmitry.kasatkin@intel.com>
Date: Thu, 26 Jan 2012 19:13:18 +0200
Subject: [PATCH] lib/mpi: added missing NULL check
Git-commit: 43b2c0aeaab2237996a72f9b9d7952ba82d56913
Patch-mainline: v3.3
References: fate#314508
Target: SLE-11 SP3

Added missing NULL check after mpi_alloc_limb_space().

Signed-off-by: Dmitry Kasatkin <dmitry.kasatkin@intel.com>
Reviewed-by: Tetsuo Handa <penguin-kernel@I-love.SAKURA.ne.jp>
Signed-off-by: James Morris <jmorris@namei.org>
Acked-by: Lee, Chun-Yi <jlee@suse.com>
---
 lib/mpi/mpi-div.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/lib/mpi/mpi-div.c
+++ b/lib/mpi/mpi-div.c
@@ -210,6 +210,8 @@ int mpi_tdiv_qr(MPI quot, MPI rem, MPI n
 		 * numerator would be gradually overwritten by the quotient limbs.  */
 		if (qp == np) {	/* Copy NP object to temporary space.  */
 			np = marker[markidx++] = mpi_alloc_limb_space(nsize);
+			if (!np)
+				goto nomem;
 			MPN_COPY(np, qp, nsize);
 		}
 	} else			/* Put quotient at top of remainder. */
